processing/src/main/java/org/apache/druid/segment/AutoTypeColumnIndexer.java

```diff
@@ -557,7 +613,14 @@ public FieldTypeInfo.MutableTypeSet getTypes()
 
     public boolean isSingleType()
     {
-      return typeSet.getSingleType() != null;
+      ColumnType singleType = typeSet.getSingleType();
```

Comment:
If the only thing seen is the empty array, then I believe that this will still return null, no?